unstract/tool-registry/src/unstract/tool_registry/tool_registry.py

```diff
@@ -45,12 +57,48 @@ def __init__(
                 "Env 'TOOL_REGISTRY_CONFIG_PATH' is not set, please add the tool "
                 "registry JSONs and YAML to a directory and set the env."
             )
+
+        if check_feature_flag_status(FeatureFlag.REMOTE_FILE_STORAGE):
+            self.fs = self._get_storage_credentials()
+        else:
+            self.fs = None
         self.helper = ToolRegistryHelper(
             registry=os.path.join(directory, registry_file),
             private_tools_file=os.path.join(directory, private_tools),
             public_tools_file=os.path.join(directory, public_tools),
+            fs=self.fs,
         )
 
+    if check_feature_flag_status(FeatureFlag.REMOTE_FILE_STORAGE):
+
+        def _get_storage_credentials(self) -> PermanentFileStorage:
+            try:
+                # Not creating constants for now for the keywords below as this
+                # logic ought to change in the near future to maintain unformity
+                # across services
+                file_storage = json.loads(
+                    os.environ.get("TOOL_REGISTRY_STORAGE_CREDENTIALS", {})
+                )
+                provider = FileStorageProvider(file_storage["provider"])
+                if "credentials" in file_storage:
+                    credentials = file_storage["credentials"]
+                else:
+                    credentials = dict()
+                return PermanentFileStorage(provider, **credentials)
+            except KeyError as e:
+                logger.error(
+                    f"Required credentials is missing" f" in the env: {str(e)}"
```

Comment:
make it with `.get` to avoid Keyerror and type issues

Reply:
Done. Thank you!